File: BookServiceJava/src/main/java/org/elcata98/mybooks/booksservice/model/Book.java
package org.elcata98.mybooks.booksservice.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;
import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

import javax.validation.constraints.NotEmpty;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class Book extends IdEntity {

    @NotEmpty
    private String title;

    @NotEmpty
    private String author;

    @NotEmpty
    private String language;

    private Short year;
    private String isbn;
    private String editorial;
    private String collection;
    private String edition;
    private Byte numVolumes;
    private Byte volume = 1;
    private String location;
    private String summary;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getLanguage() {
        return language;
    }

    public void setLanguage(String language) {
        this.language = language;
    }

    public Short getYear() {
        return year;
    }

    public void setYear(Short year) {
        this.year = year;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getEditorial() {
        return editorial;
    }

    public void setEditorial(String editorial) {
        this.editorial = editorial;
    }

    public String getCollection() {
        return collection;
    }

    public void setCollection(String collection) {
        this.collection = collection;
    }

    public String getEdition() {
        return edition;
    }

    public void setEdition(String edition) {
        this.edition = edition;
    }

    public Byte getNumVolumes() {
        return numVolumes;
    }

    public void setNumVolumes(Byte numVolumes) {
        this.numVolumes = numVolumes;
    }

    public Byte getVolume() {
        return volume;
    }

    public void setVolume(Byte volume) {
        this.volume = volume;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getSummary() {
        return summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }

    @Override
    public int hashCode() {
        return HashCodeBuilder.reflectionHashCode(this, false);
    }

    @Override
    public boolean equals(Object obj) {
        return EqualsBuilder.reflectionEquals(this, obj, false);
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: BooksService/src/main/kotlin/org/elcata98/mybooks/booksservice/model/Book.kt
package org.elcata98.mybooks.booksservice.model

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.apache.commons.lang3.builder.ToStringBuilder
import org.apache.commons.lang3.builder.ToStringStyle
import java.util.UUID

class Book {

    var bookId: String? = null
        private set

    lateinit var title: String
    lateinit var author: String
    lateinit var language: String

    var year: Short? = null
    var isbn: String? = null
    var editorial: String? = null
    var collection: String? = null
    var edition: String? = null
    var numVolumes: Byte? = null
    var volume: Byte = 1
    var location: String? = null
    var summary: String? = null

    fun generateId() {
        if (bookId == null){
            bookId = UUID.randomUUID().toString()
        }
    }

    override fun equals(other: Any?): Boolean {

        return EqualsBuilder.reflectionEquals(this, other, false)
    }

    override fun hashCode(): Int {

        return HashCodeBuilder.reflectionHashCode(this, false)
    }

    override fun toString(): String {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE)
    }
}